### PR TITLE
Cmdb ingestion

### PR DESCRIPTION
**Pull request by Gonzalo-Avalos-Ribas**
# Description

Thank you for contributing to a JupiterOne integration!

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [X] This change requires a documentation update

### General Development Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [x] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [x] My changes properly paginate the target service provider's API
- [x] My changes properly handle rate limiting of the target service provider's
      API
- [x] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [x] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [ ] I have updated the `CHANGELOG.md` file to describe my changes
- [x] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export enum ServiceNowTable {
   DATABASE_TABLES = 'sys_db_object',
   GROUP_MEMBER = 'sys_user_grmember',
   INCIDENT = 'incident',
+  sys_dictionary = 'sys_db_object',
```
Comment (adam-in-ict) [inline src/client.ts:20]: nit: Do we want to change`sys_dictionary` to be in CAPS to match the other enum entries?

File: src/client.ts [hunk 2]
```diff
@@ -75,11 +81,19 @@ export class ServiceNowClient {
   }
 
   private createRequestUrl(options: {
-    table: ServiceNowTable;
+    table: string;
```
Comment (adam-in-ict) [inline src/client.ts:84]: Is this change needed?  This looks like it's moving us away from using the `ServiceNowTable` enum values even though we've updated it to include the new `sys_dictionary` item.

File: src/getStepStartStates.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import {
+  IntegrationExecutionContext,
+  StepStartStates,
+} from '@jupiterone/integration-sdk-core';
+import { IntegrationConfig } from './types';
+import { integrationSteps } from './integrationSteps';
+
+export default function getStepStartStates(
```
Comment (adam-in-ict) [inline src/getStepStartStates.ts:8]: Would it be worth adding a unit test to make sure this properly disables the CMDB steps when the cmdb_parent value isn't included in the configuration?

File: src/steps/cmdb/converters.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        attributes: data.attributes,
+        businessUnit: data.business_unit,
+        category: data.category,
+        enviroment: data.environment,
+        ipAddress: data.ip_address,
+        macAddress: data.mac_address,
+        operationalStatus: data.operational_status,
+        serial: data.serial_number,
+        subcategory: data.subcategory,
+        warrantyExpiration: data.warranty_expiration,
```
Comment (adam-in-ict) [inline src/steps/cmdb/converters.ts:43]: Is warrantyExpiration a datetime value that we should handle similar to createdOn or updatedOn?

File: src/instanceConfigFields.ts [hunk 5]
```diff
@@ -4,6 +4,9 @@ const instanceConfigFields: IntegrationInstanceConfigFieldMap = {
   hostname: {
     type: 'string',
   },
+  cmdb_parent: {
+    type: 'string',
+  },
```
Comment (adam-in-ict) [inline src/instanceConfigFields.ts:9]: Do we want to make this an optional configuration field?

File: src/client.ts (hunk 2 above)
Comment (Gonzalo-Avalos-Ribas) [reply src/client.ts:84]: Where are going to be calling for tables that re not in the enum. Those depend on the configuration + data the user has. Adding all the tables to the enum seems too much. Some of the tables we always call regardless of config or data, those are on the enum.

File: src/getStepStartStates.ts (hunk 3 above)
Comment (Gonzalo-Avalos-Ribas) [reply src/getStepStartStates.ts:8]: Yes great suggestion! Thanks!

File: docs/jupiterone.md [hunk 6]
```diff
@@ -55,6 +57,7 @@ and assigning that read-only role to a dedicated ServiceNow user.
    - `sys_user_group`
    - `sys_user_grmember`
    - `incident`
+   - `sys_db_object`
```
Comment (adam-in-ict) [inline docs/jupiterone.md:60]: Nice documentation updates! 🔥

**Review by adam-in-ict — APPROVED** [review]
